src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/WorkspaceFactory.cs

```diff
@@ -146,10 +148,20 @@ var buildTransformBlock
                 cancellationToken: cancellationToken),
 
             buildTransformBlock.LinkTo(orderingBlock, DataflowOption.PropagateCompletion),
-
-            ProjectDataSources.JoinUpstreamDataSources(joinableTaskFactory, _projectService.Services.FaultHandler, source.ActiveConfiguredProjectSource, source.ProjectBuildRuleSource)
         });
 
+        // During solution load we run evaluations but not DTB. CPS has a feature that detects when the main thread
+        // needs DTB results and allows the DTB to run before solution load completes, in order to avoid deadlocks.
+        // While it's fine to create a dataflow subscription for build data during solution load, the joining of
+        // upstream data sources can cause CPS to accidentally think that DTB is blocking the UI thread. There's only
+        // one gate for the whole solution, so even a single project in this state will cause all projects to build.
+        // In order to avoid this, we defer joining upstream sources until after the solution has loaded.
+        _ = _tasksService.SolutionLoadedInHost.ContinueWith(
```

Comment:
Is there an advantage to using `ContinueWith` over a `FireAndForget`?

Reply:
Can you explain how you'd use `FireAndForget` here? `ContinueWith` means _run this after the task completes_, while `FireAndForget` means _run this async stuff without waiting on it_.